app/src/main/java/com/trianguloy/urlchecker/modules/list/PatternModule.java

```diff
@@ -158,7 +161,7 @@ public void onModifyUrl(UrlData urlData, JavaUtils.Function<UrlData, Boolean> se
 
                     if (replacement != null) {
                         // replace url
-                        message.newUrl = url.replaceAll(regex, replacement);
+                        message.newUrl = replaceAll(url, regex, replacement);
```

Comment:
Since this is an (impressive) hax, please move all the new logic to a new class (maybe utils.RegexpFix? Not sure).
In any case, this line here should call that class, and that's it.